source/Detector/Amptek/AmptekSDD123Detector.cpp

```diff
@@ -0,0 +1,656 @@
+//	//qDebug() << spacedStatus;
+
+	bool ok;
+	QByteArray tmpData;
+	backwardsMid(0, 4, statusData, tmpData);
+	internalSetFastCounts(tmpData.toHex().toInt(&ok, 16));
+	//qDebug() << "Fast: " << tmpData.toHex() << fastCounts_;
+
+	backwardsMid(4, 4, statusData, tmpData);
+	internalSetSlowCounts(tmpData.toHex().toInt(&ok, 16));
+	//qDebug() << "Slow: " << tmpData.toHex() << slowCounts_;
+
+	backwardsMid(8, 4, statusData, tmpData);
+	internalSetGeneralPurposeCounter(tmpData.toHex().toInt(&ok, 16));
+	//qDebug() << "gpCounter: " << tmpData.toHex() << gpCounter;
+
+	backwardsMid(12, 1, statusData, tmpData);
+	int accTime = tmpData.toHex().toInt(&ok, 16);
+	//qDebug() << "accTime: " << tmpData.toHex() << accTime;
+
+	backwardsMid(13, 3, statusData, tmpData);
+	accTime += tmpData.toHex().toInt(&ok, 16)*100;
+	internalSetAccumulationTime( ((double)(accTime))/1000);
+	//qDebug() << "accTime: " << tmpData.toHex() << accTime;
+
+	backwardsMid(16, 4, statusData, tmpData);
+	internalSetLiveTime( (double)tmpData.toHex().toInt(&ok, 16)/1000);
+	//qDebug() << "liveTime: " << tmpData.toHex() << liveTime;
+
+	backwardsMid(20, 4, statusData, tmpData);
+	internalSetRealTime( (double)tmpData.toHex().toInt(&ok, 16)/1000);
+	//qDebug() << "realTime: " << tmpData.toHex() << realTime;
+
+	backwardsMid(24, 1, statusData, tmpData);
+	tmpData[0] = tmpData[0] & 0xf0;
+	tmpData[0] = tmpData[0] >> 4;
+	internalSetFirmwareMajorVersion(tmpData.toHex().toInt(&ok, 16));
+	//qDebug() << "firmwareMajorVersion: " << tmpData.toHex() << firmwareMajorVersion;
+
+	backwardsMid(24, 1, statusData, tmpData);
+	tmpData[0] = tmpData[0] & 0x0f;
+	internalSetFirmwareMinorVersion(tmpData.toHex().toInt(&ok, 16));
+	//qDebug() << "firmwareMinorVersion: " << tmpData.toHex() << firmwareMinorVersion;
+
+	backwardsMid(25, 1, statusData, tmpData);
+	tmpData[0] = tmpData[0] & 0xf0;
+	tmpData[0] = tmpData[0] >> 4;
+	internalSetFpgaMajorVersion(tmpData.toHex().toInt(&ok, 16));
+	//qDebug() << "fpgaMajorVersion: " << tmpData.toHex() << fpgaMajorVersion;
+
+	backwardsMid(25, 1, statusData, tmpData);
+	tmpData[0] = tmpData[0] & 0x0f;
+	internalSetFpgaMinorVersion(tmpData.toHex().toInt(&ok, 16));
+	//qDebug() << "fpgaMinorVersion: " << tmpData.toHex() << fpgaMinorVersion;
+
+	backwardsMid(26, 4, statusData, tmpData);
+	internalSetSerialNumber(tmpData.toHex().toInt(&ok, 16));
+	 //qDebug() << "serialNumber: " << tmpData.toHex() << serialNumber;
+
+	tmpData = statusData.mid(30, 2);
+	internalSetHighVoltage((double)(tmpData.toHex().toUShort(&ok, 16)) * 0.5);
+	//qDebug() << "hv: " << tmpData.toHex() << shortHV << hv;
+
+	tmpData = statusData.mid(32, 2);
+	tmpData[0] = tmpData[0] & 0x0f;
+	internalSetDetectorTemperature(tmpData.toHex().toInt(&ok, 16) * 0.1);
+	//qDebug() << "detectorTemp: " << tmpData.toHex() << detectorTemp;
+
+	tmpData = statusData.mid(34, 1);
+	internalSetBoardTemperature(tmpData.at(0));
+	//qDebug() << "boardTemp: " << tmpData.toHex() << boardTemp;
+
+	tmpData = statusData.mid(35, 1);
```

Comment:
Question: is the same status data used for multiple things?  You use `mid(35, 1)` 5 times in a row for different things.